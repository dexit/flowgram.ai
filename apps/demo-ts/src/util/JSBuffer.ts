export class JSBuffer {
  buffer: string[] = []

  constructor(public indent = '') {}

  clear(): JSBuffer {
    this.buffer = []
    return this
  }

  push(...args: string[]): JSBuffer {
    this.buffer.push(...args)
    return this
  }

  pushLine(...args: string[]): JSBuffer {
    this.buffer.push(...args, '\n')
    return this
  }

  toJs(separator: string = ''): string {
    // TODO js string perf
    // FIXME last line indent
    return (
      this.buffer
        .join(separator)
        .split('\n')
        .map((line) => `${this.indent}${line}`)
        // .map((line) => (line !== '\n' ? `${this.indent}${line}` : line))
        .join('\n')
    )
  }

  toString(): string {
    return this.toJs()
  }
}
