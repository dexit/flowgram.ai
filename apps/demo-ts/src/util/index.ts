export * from './log'
export * from './JSBuffer'
