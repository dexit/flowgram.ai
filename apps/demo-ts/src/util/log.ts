import debug from 'debug'

export const info = debug('log:info')
export const log = info
export const warn = debug('log:warn')
export const error = debug('log:error')
