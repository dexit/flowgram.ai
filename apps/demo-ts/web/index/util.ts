
export const codeMirrorBaseConfig: CodeMirror.EditorConfiguration = Object.freeze({
  mode: { name: 'javascript', json: true },
  lineNumbers: true,
  lineWrapping: false,
  theme: 'twilight',
  foldGutter: true,
  gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter'],
  readOnly: true,
  cursorBlinkRate: -1,
})