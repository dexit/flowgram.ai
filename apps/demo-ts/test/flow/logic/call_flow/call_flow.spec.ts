import { runCases } from '../../util'

runCases(['call_flow'], __dirname)
