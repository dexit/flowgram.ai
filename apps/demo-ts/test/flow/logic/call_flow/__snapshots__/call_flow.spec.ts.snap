// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flow call_flow | {} 1`] = `
"export async function __main(__args) {
  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {}
  Object.assign(start.outputs, __args)
  //=== script_adder@callflow | {level:1,blockIndex:1}
  const script_adder = {}
  script_adder.outputs = {
    result: 0,
  }
  script_adder.data = {
    projectId: '96768002306',
    teamId: '0',
    a: 3.0,
    b: 4.0,
  }
  script_adder.outputs = await (
    await import(
      '/Users/bytedance/Documents/byte-proj/bwts/test/flow/logic/script/script_adder.flow'
    )
  )?.__main(script_adder.data)
  //=== end@end | {level:1,blockIndex:2}
  const end = {}
  end.data = {
    result: script_adder.outputs.result,
  }
  return end.data
}
"
`;

exports[`flow call_flow | {} 2`] = `
{
  "error": {
    "code": "",
    "msg": "",
  },
  "result": 7,
}
`;
